### PR TITLE
Feature/fuction composition

### PR DESCRIPTION
**Pull request by Adelechka**
Function Composition pattern (https://github.com/iluwatar/java-design-patterns/issues/2897)

Related: https://github.com/iluwatar/java-design-patterns/issues/2897
Pull request description

sample java code with javadoc
test code
UML diagram
README.md file

### CODE REVIEW

**Review by iluwatar — CHANGES_REQUESTED** [review]
The changes seem to contain also Virtual Proxy. Please isolate them from this PR - this should be only about Function Composition.

**Comment by Adelechka** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=iluwatar_java-design-patterns&pullRequest=2954) **Quality Gate failed**
> Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=iluwatar_java-design-patterns&pullRequest=2954&resolved=false&sinceLeakPeriod=true)  [75.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=iluwatar_java-design-patterns&pullRequest=2954&metric=new_coverage&view=list) (required ≥ 80%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=iluwatar_java-design-patterns&pullRequest=2954)

problems found in a file that is not in my pull request

File: function-composition/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+title: Function Composer
```
Comment (iluwatar) [inline function-composition/README.md:2]:
```suggestion
title: Function Composition
```

File: function-composition/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+category: Behavioral
+language: en
+tag:
+  - Functional Programming
+  - Composition
+  - Java
+---
+
+## Name / classification
```
Comment (iluwatar) [inline function-composition/README.md:11]: this section not needed, see https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template

File: function-composition/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+---
+title: Function Composer
+category: Behavioral
+language: en
+tag:
+  - Functional Programming
+  - Composition
+  - Java
```
Comment (iluwatar) [inline function-composition/README.md:8]: Prefer tags from the list https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags

File: function-composition/pom.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.20</version> <!-- Use the latest version -->
+            <scope>provided</scope>
+        </dependency>
```
Comment (iluwatar) [inline function-composition/pom.xml:52]: not needed, comes from the parent

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @Adelechka for code

**Pull request merged**